### ISSUE TITLE
no-unused-modules "missingExports" returns false positives

### ISSUE THREAD

**Issue by Slapbox** [import-js/eslint-plugin-import#2102]
**I'd expect both of these cases to count as exports:**

`module.exports = {}` // Should probably not trigger

`export * from './someFile'` // Should almost definitely not trigger

**Comment by ljharb** [comment]
`module.exports = {}` should definitely count.

`export * from './someFile'` should treat the module as if it had the same exports as `someFile`.

**Comment by Slapbox** [comment]
Similar issue:

## foo/index.js

```js
  export * as someName from './bar'
```

## index.js
  import { someName } from './foo';
```

results in "someNamed was not found in './foo'"

**Comment by Slapbox** [comment]
I just removed an eslint-directive that I noted we'd needed due to this issue, but unless ESLint is doing something weird right now, it seems like this issue may be fixed in `2.25.x`

If anyone else hits this, give the issue a bump and I'll re-open it.

**Comment by Slapbox** [comment]
> `module.exports = {}` should definitely count.

> `export * from './someFile'` should treat the module as if it had the same exports as someFile.

Unfortunately it seems neither of these cases are actually resolved as of `2.26.0`. I don't know what led me to believe they were. Perhaps I failed to re-enable the rule properly.

@ljharb any thoughts about possible causes and solutions?

**Comment by Slapbox** [comment]
Our `settings` key in `.eslintrc.js` looks like this - I wonder if it's possibly related to resolution issues? But it seems to work fine in other capacities, so I'm doubtful.

```js
settings: {
    // 'import/resolver': 'webpack',
    'import/cache': { lifetime: 0 },
    'import/external-module-folders': ['node_modules', 'packages'],
    'import/resolver': {
      'babel-module': {},
      webpack: {
        config: './webpack_config/webpack.config.resolve.js',
      },
    },
  },
```

**Comment by ljharb** [comment]
@Slapbox it might be that you're missing the node resolver - add that last, after webpack.

**Comment by Slapbox** [comment]
Sorry for the lengthy absence here.

I finally got to give that a try but unfortunately it makes no difference.

I added 

```js
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
```

And installed `eslint-import-resolver-node`
